src/Dialog/DialogBottomSheet.tsx

```diff
@@ -0,0 +1,327 @@
+import React, {useEffect, useRef, useState, PropsWithChildren, MouseEvent, TouchEvent, ChangeEvent} from 'react'
+import styled from 'styled-components'
+import {SxProp} from '../sx'
+import {useRefObjectAsForwardedRef} from '../hooks/useRefObjectAsForwardedRef'
+import Box from '../Box'
+import {get} from '../constants'
+
+const ANIMATION_DURATION = 300
+const FULL_HEIGHT = 95
+const HALF_HEIGHT = 50
+
+/**
+ * Props to customize the rendering of the Dialog.
+ */
+export interface DialogBottomSheetProps extends SxProp {
+  /**
+   * This method is invoked when a gesture to close the dialog is used
+   */
+  onClose: (gesture: 'close-button' | 'escape' | 'drag' | 'overlay') => void
+
+  /**
+   * Passed through to use as the draggable area of the dialog.
+   */
+  header: React.ReactNode
+
+  /**
+   * Default: "dialog". The ARIA role to assign to this dialog.
+   * @see https://www.w3.org/TR/wai-aria-practices-1.1/#dialog_modal
+   * @see https://www.w3.org/TR/wai-aria-practices-1.1/#alertdialog
+   */
+  role?: 'dialog' | 'alertdialog'
+
+  /**
+   * Identifies the element (or elements) that labels the element it is applied to.
+   */
+  ariaLabelledby: string
+
+  /**
+   * Identifies the element (or elements) that describes the element on which the attribute is set.
+   */
+  ariaDescribedby: string
+
+  /**
+   * Indicates whether an element is modal when displayed.
+   */
+  ariaModal: boolean
```

Comment:
This seems right to me, but I'd double-check with a11y folks. Some of the `aria-` attributes have inconsistent behavior on different screen readers.

Reply:
@ichelsea @ericwbailey is that something you could double check for me?

Reply:
Also tagging @alexislucio since she's this week's a11y FR.